app/vmctl/vm_native.go

```diff
@@ -52,10 +54,52 @@ const (
 )
 
 func (p *vmNativeProcessor) run(ctx context.Context) error {
+	if p.filter.chunk == "" {
+		return p.runSingle(ctx, p.filter)
+	}
+
+	startOfRange, err := time.Parse(time.RFC3339, p.filter.timeStart)
+	if err != nil {
+		return fmt.Errorf("failed to parse %s, provided: %s, expected format: %s, error: %v", vmNativeFilterTimeStart, p.filter.timeStart, time.RFC3339, err)
+	}
+
+	var endOfRange time.Time
+	if p.filter.timeEnd != "" {
+		endOfRange, err = time.Parse(time.RFC3339, p.filter.timeEnd)
+		if err != nil {
+			return fmt.Errorf("failed to parse %s, provided: %s, expected format: %s, error: %v", vmNativeFilterTimeEnd, p.filter.timeEnd, time.RFC3339, err)
+		}
+	} else {
+		endOfRange = time.Now()
+	}
+
+	ranges, err := splitDateRange(startOfRange, endOfRange, p.filter.chunk)
+	if err != nil {
+		return fmt.Errorf("failed to create date ranges for the given time filters: %v", err)
+	}
+
+	for rangeIdx, r := range ranges {
+		log.Printf("Processing range %d/%d: %s - %s \n", rangeIdx+1, len(ranges), r.start.Format(time.RFC3339), r.end.Format(time.RFC3339))
```

Comment:
nit: we can define local variables like `formattedStartTime` and `formattedEndTime` or similar and use them in the next part of code instead of call Format into different places

Reply:
Refactored to address your suggestionsbb